### PR TITLE
Add CI (GitHub Actions)

### PR DESCRIPTION
**Pull request by myfreeweb**
In order to reduce issues between different C/C++ stacks, let's continuously test on a couple different ones. A mostly-GNU-but-musl one and a mostly-LLVM-but-glibc one was the best I could come up with. Here's [an example run](https://github.com/myfreeweb/wayfire/actions/runs/41498661).

- DEPENDS ON https://github.com/WayfireWM/wf-config/pull/21
- would be cool to also run static analysis in CI I guess (but where would it upload the nice HTML results?)
- maybe also test [wf-gsettings](https://github.com/myfreeweb/wf-gsettings) xml conversion :D
- maybe also enforce clang-format (when that happens) in CI (like Mozilla does for Firefox)
- this PR also fix some things found by clang analyzer (not tested running with these yet.. I don't run anything using wayfire-shell anyway..), I should probably split that off

### CODE REVIEW

**Comment by ammen99** [comment]
> * maybe also enforce clang-format (when that happens) in CI (like Mozilla does for Firefox)

+1

File: src/output/wayfire-shell.cpp [hunk 1]
```diff
@@ -190,7 +190,6 @@ class wfs_output : public noncopyable_t
         wf::get_core().output_layout->disconnect_signal("output-removed",
             &on_output_removed);
         output->disconnect_signal("fullscreen-layer-focused", &on_fullscreen_layer_focused);
-        this->output = nullptr;
```
[the remark this answers is not in the document]
Comment (myfreeweb) [reply src/output/wayfire-shell.cpp]: Feel free to fix it however you want, I haven't used `wayfire-shell` so I have no idea about the logic here :)

**Comment by ammen99** [comment]
By the way, I saw you already have a second PR opened but this one is WIP, any particular reason?

**Comment by myfreeweb** [comment]
I mostly wasn't sure about the extra crap I threw in here, I'll keep this CI only now that the wf shell issue is known.

**Comment by ammen99** [comment]
I think there was a nice badge for the README :P 

Also, I think you should submit the clang-analyzer things, I'll review them as well and maybe @soreau can add more commits if necessary.

**Comment by myfreeweb** [comment]
> I think there was a nice badge for the README :P

Added (along with other badge things haha). CI badge won't work until this is merged, but you can preview it by replacing `WayfireWM` with `myfreeweb` in the image URL.

> submit the clang-analyzer things

Well, I think my decorations cache thing might have introduced crashes, and @soreau can fix the wayfire shell thing in a different way :)

**Comment by ammen99** [comment]
Thanks for this, I am not very familiar with the Github CI but if it runs, it runs :)

**Pull request merged**